File: Languages/R/Day012.r
# Languages - R - Day 12
# Foundational Concepts: Basics, syntax, loops, conditionals, functions, and variables.

# Create a function that takes two numbers as parameters and returns the larger of the two. Call this function and print the result.

# Function to find the larger of two numbers
find_larger <- function(num1, num2) {
  if (num1 > num2) {
    return(num1)
  } else {
    return(num2)
  }
}

# Function to validate user input and ensure it is numeric
get_valid_number <- function(prompt) {
  repeat {
    input <- readline(prompt = prompt)
    num <- as.numeric(input)
    if (!is.na(num)) {
      return(num)
    } else {
      cat("Invalid input. Please enter a valid number.\n")
    }
  }
}

# Get two valid numbers from the user
num1 <- get_valid_number("Enter the first number: ")
num2 <- get_valid_number("Enter the second number: ")

# Call the function and print the result
result <- find_larger(num1, num2)
cat("The larger number is:", result, "\n")

File: Languages/C#/Day012.cs
// Languages - C# - Day 12
// Foundational Concepts: Basics, syntax, loops, conditionals, functions, and variables.

// Create a function that takes two numbers as parameters and returns the larger of the two. Call this function and print the result.

using System;

class Program
{
    static void Main(string[] args)
    {
        // Get two valid numbers from the user
        int num1 = GetValidNumber("Enter the first number: ");
        int num2 = GetValidNumber("Enter the second number: ");

        // Call the function and print the result
        int largerNumber = GetLargerNumber(num1, num2);
        Console.WriteLine("The larger number is: " + largerNumber);
    }

    // Function to find the larger of two numbers
    static int GetLargerNumber(int a, int b)
    {
        return (a > b) ? a : b;
    }

    // Function to validate and get a numeric input from the user
    static int GetValidNumber(string prompt)
    {
        int number;
        while (true)
        {
            Console.Write(prompt);
            string input = Console.ReadLine();
            if (int.TryParse(input, out number))
            {
                return number;
            }
            else
            {
                Console.WriteLine("Invalid input. Please enter a valid integer.");
            }
        }
    }
}
